### PR TITLE
Update ClientKey and AuthMiddleware for v2 API work

### PR DESCRIPTION
**Pull request by haitaottd**

### CODE REVIEW

File: src/main/java/com/uid2/shared/auth/ClientKey.java [hunk 1]
```diff
@@ -48,13 +51,15 @@ public class ClientKey implements IRoleAuthorizable<Role> {
     public ClientKey(String key, String secret) {
         this.key = key;
         this.secret = secret;
+        this.secretBytes = Utils.decodeBase64String(secret);
```
Comment (atarassov-ttd) [inline src/main/java/com/uid2/shared/auth/ClientKey.java:54]: Should this call `setSecret()` to follow the DRY principle?

File: src/main/java/com/uid2/shared/middleware/AuthMiddleware.java [hunk 2]
```diff
@@ -198,6 +214,24 @@ public void handle(RoutingContext rc) {
             }
         }
 
+        public void handleV2(RoutingContext rc, Role role) {
+            // add aws request id tracer to help validation
```
Comment (atarassov-ttd) [inline src/main/java/com/uid2/shared/middleware/AuthMiddleware.java:218]: This looks like a copy paste from above and does not really belong to auth logic. Can this be extracted into a re-usable method or, even better, into a handler of its own?

File: src/main/java/com/uid2/shared/middleware/AuthMiddleware.java [hunk 3]
```diff
@@ -198,6 +214,24 @@ public void handle(RoutingContext rc) {
...
+            String traceId = rc.request().getHeader("X-Amzn-Trace-Id");
+            if (traceId != null && traceId.length() > 0) {
+                rc.response().headers().add("X-Amzn-Trace-Id", traceId);
+            }
+
+            final String authHeaderValue = rc.request().getHeader(AuthMiddleware.AuthorizationHeader);
+            final String authKey = AuthHandler.extractBearerToken(authHeaderValue);
+            final ClientKey profile = (ClientKey)this.authKeyStore.get(authKey);
+            AuthMiddleware.setAuthClient(rc, profile);
+            if (profile == null || profile.isDisabled() || !profile.getRoles().contains(role)) {
```
Comment (atarassov-ttd) [inline src/main/java/com/uid2/shared/middleware/AuthMiddleware.java:228]: What is the reason for not using the `authorizationProvider` here? How is this method different from the `handle() method above?

File: src/main/java/com/uid2/shared/auth/ClientKey.java (hunk 1 above)
Comment (haitaottd) [reply src/main/java/com/uid2/shared/auth/ClientKey.java:54]: done

File: src/main/java/com/uid2/shared/middleware/AuthMiddleware.java (hunk 2 above)
Comment (haitaottd) [reply src/main/java/com/uid2/shared/middleware/AuthMiddleware.java:218]: Reverted. Thought v2 auth may be different.

File: src/main/java/com/uid2/shared/middleware/AuthMiddleware.java (hunk 3 above)
Comment (haitaottd) [reply src/main/java/com/uid2/shared/middleware/AuthMiddleware.java:228]: Reverted. authorizationProvider seems a little overkill.

**Review by atarassov-ttd — APPROVED** [review]
LGTM

**Pull request merged**